lib/db/lowlevel.go

```diff
@@ -201,7 +201,7 @@ func (db *Lowlevel) updateLocalFiles(folder []byte, fs []protocol.FileInfo, meta
 		blocksHashSame := ok && bytes.Equal(ef.BlocksHash, f.BlocksHash)
 
 		if ok {
-			if !ef.IsDirectory() && !ef.IsDeleted() && !ef.IsInvalid() {
+			if len(ef.Blocks) != 0 {
```

Comment:
Can't something invalid (like a locally modified receive only file or something) not have blocks?

Reply:
It's not supposed to, but keeping the IsInvalid() check might make sense